hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -656,7 +657,12 @@ private ReferenceCounted<OmSnapshot> getSnapshot(
       // don't allow snapshot indicator without snapshot name
       throw new OMException(INVALID_KEY_NAME);
     }
-
+    // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
+    // permission check, since linked bucket permissions and source bucket permissions could be different.
+    ResolvedBucket resolvedBucket = ozoneManager.resolveBucketLink(Pair.of(volumeName,
```

Comment:
Do we need to have it here? I think we should change it on the entry point in OzoneManager for the read operations and others in the corresponding `preExecute`.

Reply:
No we should not do that. That would impact all operations

Reply:
We don't want bucket resolution for other operations excepting for the snapshot operations

Reply:
Then it could be just here
https://github.com/apache/ozone/blob/2a3c55aaa4804e0c033d1a35b90630e4f2f1471b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java#L616

I'll leave it on you.

Reply:
Oh I am sorry I didn't see the file correctly before. I have made the same change based on @sumitagrawl comments